File: easy_video/easyvideo_video_ui/src/api/video/chapter.js
/*
 * Lan Sheng  Copyright (c) 2022 https://easyvideo.top.
 * Copyright (c) 2022. easyvideo, personal projects are not allowed to be commercialized without permission.
 * Please keep the information of the original author of the code. Thank you
 */

import request from '@/utils/request'

const apiName = '/service_video/chapter'
export default {
  getNestedTreeList(contentId) {
    return request({
      url: `${apiName}/getChapterContentVideo/${contentId}`,
      method: 'get'
    })
  },
  insertChapter(params) {
    return request({
      url: `${apiName}`,
      method: 'post',
      data: params
    })
  },
  updateChapter(params) {
    return request({
      url: `${apiName}`,
      method: 'put',
      data: params
    })
  },
  deleteChapterById(id) {
    return request({
      url: `${apiName}/${id}`,
      method: 'delete'
    })
  },
  getContentOrVideoSortMax(type) {
    return request({
      url: `${apiName}/getContentOrVideoSortMax/${type}`,
      method: 'post'
    })
  }
}

File: easy_video/easyvideo_api/api_video/src/main/java/top/easyvideo/api_video/entity/Chapter.java
// Lan Sheng  Copyright (c) 2022 https://127.0.0.1.
// Copyright (c) 2022. easyvideo, personal projects are not allowed to be commercialized without permission.
// Please keep the information of the original author of the code. Thank you

package top.easyvideo.api_video.entity;

import com.baomidou.mybatisplus.annotation.*;

import java.util.Date;
import java.io.Serializable;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 作品章节
 * </p>
 *
 * @author easyvideo
 * @since 2021-09-19
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("video_chapter")
@ApiModel(value="Chapter对象", description="作品章节")
public class Chapter implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "章节ID")
    @TableId(value = "id", type = IdType.ID_WORKER_STR)
    private String id;

    @ApiModelProperty(value = "作品ID")
    private String contentId;

    @ApiModelProperty(value = "章节名称")
    private String title;

    @ApiModelProperty(value = "显示排序")
    private Integer sort;

    @TableField(value = "`gmt_create`",fill = FieldFill.INSERT)
    @ApiModelProperty(value="创建时间")
    private Date gmtCreate;

    @TableField(value = "`gmt_modified`",fill = FieldFill.INSERT_UPDATE)
    @ApiModelProperty(value="更新时间")
    private Date gmtModified;


}
